actors.md

```diff
@@ -4,6 +4,42 @@ Any implementations of the Filecoin actors must be exactly byte for byte compati
 
 This spec decsribes a set of actors that operate within the [Filecoin State Machine](state-machine.md). All types are defined in [the basic type encoding spec](data-structures.md#basic-type-encodings).
 
+## Init Actor
+The init actor is responsible for creating new actors on the filecoin network. This is a built-in actor and cannot be replicated. In the future, this actor will be responsible for loading new code into the system (for user programmable actors).
+
+```go
+type InitActor struct {
+    NextActorID uint64
+}
+```
+
+#### Exec
+
+Parameters:
+
+- code Cid
+- params []byte
+
+TODO: the type of params should probably just be variadic set of arguments. Making it be a `[]byte` here kinda sucks
+
+```go
+func Exec(code Cid, params []byte) Address {
+    actorID := self.NextActorID
+    self.NextActorID++
+    
+    if code == InitActorCodeCid {
+        Fatal("cannot launch additional init actor instance")
+    }
+    
+    addr := VM.CreateActor(code, msg.Value, DecodeParams(params))
+    
+    // TODO: where is this mapping stored?
```

Comment:
might as well store it in this actor?

Reply:
yeah... that doesnt seem like the worst thing to do. I'll come up with some pros and cons in a bit